server/src/main/java/org/opensearch/search/query/QueryPhaseSearcher.java

```diff
@@ -41,7 +41,7 @@ boolean searchWith(
         LinkedList<QueryCollectorContext> collectors,
         boolean hasFilterCollector,
         boolean hasTimeout
-    ) throws IOException;
+    ) throws Throwable;
```

Comment:
Good catch! Let me take a look at how I can fix this is `ConcurrentQueryPhaseSearcher::searchWithCollectorManager` then

Reply:
From what I can see, if we want to avoid throwing an unchecked exception then we can't directly throw `RuntimeException.getCause()`, we would have to check `instanceof` first and then cast it to the right exception type. That's not difficult to do, but it feels kind of hacky. Do you have any other thoughts @sohami @reta ?

Reply:
Updated `ConcurrentQueryPhaseSearcher::searchWithCollectorManager` in the latest revision with this change to give a better idea of what I'm describing here

Reply:
I suppose we could also just add `@SuppressWarnings("unchecked")` too but I don't like that solution either